pkg/operator/workload/workload_openshiftapiserver_v311_00_sync.go

```diff
@@ -273,15 +279,45 @@ func manageOpenShiftAPIServerImageImportCA_v311_00_to_latest(ctx context.Context
 	return resourceapply.ApplyConfigMap(ctx, client, recorder, requiredConfigMap)
 }
 
-func manageOpenShiftAPIServerConfigMap_v311_00_to_latest(ctx context.Context, client coreclientv1.ConfigMapsGetter, recorder events.Recorder, operatorConfig *operatorv1.OpenShiftAPIServer) (*corev1.ConfigMap, bool, error) {
+func manageOpenShiftAPIServerConfigMap_v311_00_to_latest(ctx context.Context, client coreclientv1.ConfigMapsGetter, clusterVersionInformer configinformersconfigv1.ClusterVersionInformer, recorder events.Recorder, operatorConfig *operatorv1.OpenShiftAPIServer) (*corev1.ConfigMap, bool, error) {
 	configMap := resourceread.ReadConfigMapV1OrDie(v311_00_assets.MustAsset("v3.11.0/openshift-apiserver/cm.yaml"))
 	defaultConfig := v311_00_assets.MustAsset("v3.11.0/config/defaultconfig.yaml")
+
+	clusterVersion, err := clusterVersionInformer.Lister().Get("version")
+	if err != nil {
+		return nil, false, err
+	}
+
+	knownCaps := sets.New[configv1.ClusterVersionCapability](clusterVersion.Status.Capabilities.KnownCapabilities...)
+	capsEnabled := sets.New[configv1.ClusterVersionCapability](clusterVersion.Status.Capabilities.EnabledCapabilities...)
+
+	apiServersConfig := openshiftcontrolplanev1.OpenShiftAPIServerConfig{
+		APIServers: openshiftcontrolplanev1.APIServers{
+			PerGroupOptions: []openshiftcontrolplanev1.PerGroupOptions{},
+		},
+	}
+	if knownCaps.Has(configv1.ClusterVersionCapabilityBuild) && !capsEnabled.Has(configv1.ClusterVersionCapabilityBuild) {
```

Comment:
Will you have a test that will test the disabling of this APIs?

Reply:
Already in https://github.com/openshift/cluster-openshift-apiserver-operator/pull/532/commits/dc8a55f7c7352a5d0f1e8599a042f13aa913cab6#diff-25e7c9a2521d15c93e122ab635c1ec19358116650aa275abca5e014b514c69c2R241

Reply:
I think that we should have an e2e test for it, wdyt?

Reply:
Updating/adding e2es is the next step once this PR and https://github.com/openshift/cluster-openshift-controller-manager-operator/pull/291 are merged.

Reply:
where are you planing to add that test ? will it be this repo or in the origin ?

Reply:
Origin as we need to check for disabled controllers as well. E.g. read the config or parse the controllers logs.

Reply:
Ideally if we could add a test here and then also run it from the origin repo. Are you planing to add the test before we ship 4.14 ?

Reply:
Yes. All tests need to land before 4.14 ships.

Reply:
OK. I would like to avoid shipping a feature without the tests. Thanks.